examples/website/terrain-extension/app.jsx

```diff
@@ -0,0 +1,105 @@
+/* eslint-disable max-statements */
+import React from 'react';
+import {createRoot} from 'react-dom/client';
+import DeckGL from '@deck.gl/react';
+
+import {TerrainLayer} from '@deck.gl/geo-layers';
+import {GeoJsonLayer, TextLayer} from '@deck.gl/layers';
+import {_TerrainExtension as TerrainExtension} from '@deck.gl/extensions';
+import data from './data/data';
+
+// Set your mapbox token here
+const MAPBOX_TOKEN = process.env.MapboxAccessToken;
+
+const INITIAL_VIEW_STATE = {
+  latitude: 43.09822,
+  longitude: -0.6194,
+  zoom: 10,
+  pitch: 55,
+  maxZoom: 13.5,
+  bearing: 0,
+  maxPitch: 89
+};
+
+const TERRAIN_IMAGE = `https://api.mapbox.com/v4/mapbox.terrain-rgb/{z}/{x}/{y}.png?access_token=${MAPBOX_TOKEN}`;
+const SURFACE_IMAGE = `https://api.mapbox.com/v4/mapbox.satellite/{z}/{x}/{y}@2x.png?access_token=${MAPBOX_TOKEN}`;
+
+// https://docs.mapbox.com/help/troubleshooting/access-elevation-data/#mapbox-terrain-rgb
+// Note - the elevation rendered by this example is greatly exagerated!
+const ELEVATION_DECODER = {
+  rScaler: 6553.6,
+  gScaler: 25.6,
+  bScaler: 0.1,
+  offset: -10000
+};
+
+const COLOR_SCHEME = [242, 183, 1]; // yellow
+
+function getTooltip({object}) {
+  return (
+    object && {
+      html: `\
+  <div><b>Stage</b></div>
+  <div>${object.properties.name}</div>
+  `
+    }
+  );
+}
+
+export default function App({initialViewState = INITIAL_VIEW_STATE}) {
+  const labels = data.features.filter(f => f.geometry.type === 'Point');
+  const layers = [
+    new TerrainLayer({
+      id: 'terrain',
+      minZoom: 0,
+      strategy: 'no-overlap',
+      elevationDecoder: ELEVATION_DECODER,
+      elevationData: TERRAIN_IMAGE,
+      texture: SURFACE_IMAGE,
+      wireframe: false,
+      color: [255, 255, 255],
+      operation: 'terrain+draw'
+    }),
+    new GeoJsonLayer({
+      id: 'terrain-routes',
+      data,
+      getLineColor: () => COLOR_SCHEME,
+      getFillColor: () => COLOR_SCHEME,
+      getLineWidth: 20,
+      stroked: false,
+      getPointRadius: 75,
+      lineWidthMinPixels: 2,
+      pointType: 'circle',
+      pickable: true,
+      autoHighlight: true,
+      highlightColor: [255, 200, 0],
+      extensions: [new TerrainExtension()]
+    }),
+    new TextLayer({
```

Comment:
Why is a separate layer needed? Can you just use `pointType: 'text'` in the GeoJsonLayer?